### ISSUE TITLE
Aborting requests throws a ReferenceError in node.js

### ISSUE THREAD

**Issue by solmsted** [JMPerez/promise-throttle#45]
Using node.js v8.11.1

Simple example to reproduce:

```js
const PromiseThrottle = require('promise-throttle'),
    promiseThrottle = new PromiseThrottle({
        promiseImplementation: Promise,
        requestsPerSecond: 1
    });

promiseThrottle.add(() => Promise.resolve(), {
    signal: {
        aborted: true
    }
});
```

This results in an error: `UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): ReferenceError: DOMException is not defined`

**Comment by JMPerez** [comment]
You are right, @solmsted. As far as I know, `AbortSignal` and `AbortController` are not supported in Node.JS. One reason is that `DOMException` is not defined in Node.JS.

[According to the spec](https://dom.spec.whatwg.org/#aborting-ongoing-activities):

> APIs that rely upon AbortController are encouraged to respond to abort() by rejecting any unsettled promise with a new "AbortError" DOMException.

At this point, the library either triggers an early exception when trying to use a `signal` in node, or it throws a regular `Exception` when aborting in node.

**Comment by execptionerror** [comment]
[[Already Solved] UnhandledPromiseRejectionWarning: ReferenceError: AbortController is not defined at RequestHandler.execute](https://flutterq.com/unhandledpromiserejectionwarning-referenceerror-abortcontroller-is-not-defined-at-requesthandler-execute/)
